web-share/src/js/contentIndexing.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import {cachedMediaMetadataPromise} from './getCachedMediaMetadata';
+import {iconSrcs} from './constants';
+
+// This method syncs the currently cached media with the Content Indexing API
+// (on browsers that support it). The Cache Storage is the source of truth.
+export async function syncContentIndex(registration) {
+  //  Bail early if the Content Indexing API isn't supported.
+  if (!('index' in registration)) {
+    return;
+  }
+
+  // Get a list of everything currently in the content index. The id of each
+  // entry will match the media URLs that are stored in the cache.
+  const allEntries = await registration.index.getAll();
+  const idsInIndex = new Set(allEntries.map((entry) => entry.id));
+
+  // Get a list of all cached media.
+  const cachedMediaMetadata = await cachedMediaMetadataPromise;
+
+  for (const metadata of cachedMediaMetadata) {
+    if (idsInIndex.has(metadata.src)) {
+      // If a given id/URL is in both the content index and currently in our
+      // cached, remove it from the set.
+      idsInIndex.delete(metadata.src);
```

Comment:
maybe use `continue` to avoid indentation and make code easier to read.

Reply:
I'd prefer not to? Both clauses are equally possible and important, and I'd rather just have an if/else.